### ISSUE TITLE
[UserProjectIntegrator] Missing removal of Pods aggregate target product

### ISSUE THREAD

**Issue by mrackwitz** [CocoaPods/CocoaPods#4327]
## Steps to reproduce

Have a Podfile like:

```ruby
platform :ios, '8.0'
use_frameworks!

pod 'RealmSwift'
pod 'MBProgressHUD'
pod 'Alamofire'
```

Run `pod install`.

Change the Podfile to:

```ruby
platform :ios, '8.0'
use_frameworks!

def shared_pods
   pod 'RealmSwift'
   pod 'Alamofire'
end

target 'App' do
    shared_pods
    pod 'MBProgressHUD'
end

target 'AppToday' do
    shared_pods
end
```

Run `pod install` again.
Trying to build the project will yield an error `ld: framework not found Pods`.

That's because the integration will look like:

[![Both aggregate target products: the old to remove and the new to keep][1]][1]

While the integration should look like that:

[![Just the new aggregate target product][2]][2]

  [1]: http://i.stack.imgur.com/P0YCO.png
  [2]: http://i.stack.imgur.com/Lfk30.png

**Comment by segiddins** [comment]
This should be fixed now in 1.0?

**Comment by manuyavuz** [comment]
Yeah, regarding we do not have implicit targets anymore and have `deintegrate` included inside the default install flow. We can close it i guess?
